tests/test_diff.py

```diff
@@ -0,0 +1,19 @@
+from collections import deque
+
+from codemodder.diff import calc_line_num_changes
+
+
+def test_calc_line_nums_remove_line():
```

Comment:
This is a very nice test but it might be nice to see a few other cases:
* one where a line is changed (with both `+` and `-`)
* one where a new line is added (just `+`)
* one where multiple line changes occur within the same hunk

Reply:
I didn't add these mostly bc it's already tested via other non-obvious ways but happy to do so!